src/gui/sdlmain.cpp

```diff
@@ -1744,19 +1739,22 @@ static SDL_Window * SetDefaultWindowMode()
 	sdl.draw.height = splash_image.height;
 
 	if (sdl.desktop.fullscreen) {
+		sdl.desktop.lazy_init_window_size = true;
 		return GFX_SetSDLWindowMode(sdl.desktop.full.width,
 		                            sdl.desktop.full.height,
-		                            sdl.desktop.fullscreen,
+		                            true,
 		                            sdl.desktop.want_type);
 	}
 
 	if (sdl.desktop.window.use_original_size) {
+		sdl.desktop.lazy_init_window_size = false;
 		return GFX_SetSDLWindowMode(sdl.draw.width,
 		                            sdl.draw.height,
 		                            false,
 		                            sdl.desktop.want_type);
 	}
 
+	sdl.desktop.lazy_init_window_size = false;
 	return GFX_SetSDLWindowMode(sdl.desktop.window.width,
 	                            sdl.desktop.window.height,
 	                            false,
```

Comment:
This section is another example where `sdl.desktop.window.<dims>` is a second-rate citizen.  When `use_original_size` is true, `sdl.desktop.window.<dims>` is never used to define the window resolution, nor is it updated to have the current Windowed-mode resolution; (so in these cases, the only way to discover the current window-mode resolution is to directly query SDL). 

Suggestion: when we parse the `windowresolution` config file, we make sure `sdl.desktop.window.<dims>` is populated no matter what.  So even if it's set to `original`, we populate the initial Window dimensions with `splash_image.<dims>`; and then we can use those here without needing special handling.

Reply:
I mentioned in my response above why we can't do that :) Basically, it would make option `windowresolution=original` completely meaningless - which would be actually ok for most (or almost all) games, but not ok e.g. for Windows 3.11.

We might do that after 0.75 though, as it would considerably simplify the code (and make the emulator behave more predictably for the users).